pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -549,18 +553,66 @@ func (os *OpenStack) GetZone() (cloudprovider.Zone, error) {
 	return zone, nil
 }
 
+func (os *OpenStack) getZoneByServer(server *servers.Server) (cloudprovider.Zone, error) {
+	az := ""
+	var ok bool
+
+	if az, ok = server.Metadata[AvailabilityZone]; !ok {
+		return cloudprovider.Zone{}, fmt.Errorf("Can not get %s from the Metadata of instance %s", AvailabilityZone, server.Name)
```

Comment:
`nil` should be ok for the return value right? (Same below)

Reply:
for the first parameter

Reply:
Sorry, nil should be ok, If az is empty, just let caller handle it. Do not return err like GetZone().